x-pack/plugins/security_solution/common/utils/risk_score_modules.ts

```diff
@@ -0,0 +1,584 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RiskScoreEntity, RiskScoreFields } from '../search_strategy';
+
+/**
+ * * Since 8.5, all the transforms, scripts,
+ * and ingest pipelines (and dashboard saved objects) are created with spaceId
+ * so they won't affect each other across different spaces.
+ */
+export const getRiskScorePivotTransformId = (
+  riskScoreEntity: RiskScoreEntity,
+  spaceId = 'default'
+) => `ml_${riskScoreEntity}riskscore_pivot_transform_${spaceId}`;
+
+export const getRiskScoreLatestTransformId = (
+  riskScoreEntity: RiskScoreEntity,
+  spaceId = 'default'
+) => `ml_${riskScoreEntity}riskscore_latest_transform_${spaceId}`;
+
+export const getIngestPipelineName = (riskScoreEntity: RiskScoreEntity, spaceId = 'default') =>
+  `ml_${riskScoreEntity}riskscore_ingest_pipeline_${spaceId}`;
+
+export const getPivotTransformIndex = (riskScoreEntity: RiskScoreEntity, spaceId = 'default') =>
+  `ml_${riskScoreEntity}_risk_score_${spaceId}`;
+
+export const getLatestTransformIndex = (riskScoreEntity: RiskScoreEntity, spaceId = 'default') =>
+  `ml_${riskScoreEntity}_risk_score_latest_${spaceId}`;
+
+export const getRiskScoreLevelScriptId = (riskScoreEntity: RiskScoreEntity, spaceId = 'default') =>
+  `ml_${riskScoreEntity}riskscore_levels_script_${spaceId}`;
+export const getRiskScoreInitScriptId = (riskScoreEntity: RiskScoreEntity, spaceId = 'default') =>
+  `ml_${riskScoreEntity}riskscore_init_script_${spaceId}`;
+export const getRiskScoreMapScriptId = (riskScoreEntity: RiskScoreEntity, spaceId = 'default') =>
+  `ml_${riskScoreEntity}riskscore_map_script_${spaceId}`;
+export const getRiskScoreReduceScriptId = (riskScoreEntity: RiskScoreEntity, spaceId = 'default') =>
+  `ml_${riskScoreEntity}riskscore_reduce_script_${spaceId}`;
+
+/**
+ * These scripts and Ingest pipeline were not space awared before 8.5.
+ * They were shared across spaces and therefore affected each other.
+ * New scripts and ingest pipeline are all independent in each space, so these ids
+ * are Deprecated.
+ * But We still need to keep track of the old ids, so we can delete them during upgrade.
+ */
+export const getLegacyIngestPipelineName = (riskScoreEntity: RiskScoreEntity) =>
```

Comment:
Since these weren't space aware before 8.5, says user upgrades in space A, would the upgrade run into errors in a different space B if it's unable to find the legacy artifacts?

Reply:
Yes, the transforms in space B won't be able to find the scripts and ingest pipelines it needs, but anyway the space B UI will be showing  `upgrade panel` as well